go/obscuronode/enclave/l1_blockchain.go

```diff
@@ -0,0 +1,122 @@
+package enclave
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/ethereum/go-ethereum/consensus"
+	"github.com/ethereum/go-ethereum/consensus/beacon"
+	"github.com/ethereum/go-ethereum/consensus/ethash"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/eth/ethconfig"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+// NewL1Blockchain creates a Geth BlockChain object. `genesisJSON` is the Genesis block config in JSON format. A Geth
+// node can be made to output this using the `dumpgenesis` startup command.
+func NewL1Blockchain(genesisJSON []byte) *core.BlockChain {
+	dataDir, err := ioutil.TempDir(os.TempDir(), "")
```

Comment:
this should be a config with a good default value.

Reply:
I've added a default value. The configuration of it can wait on the config framework I guess?